### PR TITLE
Draft: Remaining Items for UL

### PR DESCRIPTION
**Pull request by JHiltbrand**
- Currently adds filelists and configs for data samples for all years. Also fix 2018 data scenario to use "PAT" rather than "RECO".

- Add unit tests for data and GJets.

- The `pileupCorr.py` file has been modified to read the latest versions of the pileup pdfs from central. The corresponding up and down variations of the minimum bias cross section are provided in separate ROOT files and the script has been modified to read them in as such.

- Update b tag WP for both "versions" of 2016.

- Add various TT + X, W + Jets, ZJetsToNuNu, QCD_HT, GJets, and WJetsToQQ samples.

- remove some residual reference to `wrong_pu` option that is no longer used.

- update some GTs for the different scenarios based on p.20 of https://indico.cern.ch/event/1104487/contributions/4646705/attachments/2361511/4031368/21-12-09_News_PPD.pdf

- add five DeepFlavour variables

### CODE REVIEW

File: Production/test/pileupCorr.py [hunk 1]
```diff
@@ -69,10 +69,17 @@ def DtoF(file,get):
         hdata_up.SetName("data_pu_up")
         hdata_down.SetName("data_pu_down")
     else:
-        dfile = TFile.Open(options.data)
-        hdata_central = dfile.Get("data_pu_central")
-        hdata_up = dfile.Get("data_pu_up")
-        hdata_down = dfile.Get("data_pu_down")
+        dfile  = TFile.Open(options.data, "READ") # nominal with min bias xsec 69200 ub
+        dfiled = TFile.Open(options.data.replace("69200", "66000"), "READ") # down variation with min bias xsec 66000 ub
```
Comment (kpedro88) [inline Production/test/pileupCorr.py:73]: Can you add some more details about why this change is needed?

File: Production/test/unitTest.py [hunk 2]
```diff
@@ -93,12 +93,22 @@ def defineTests(mytests, scenario, name, numevents, command, dataset, inputFiles
     mytests.append(Test(scenario,name,numevents,command,dataset,inputFilesConfig,nstart=0,nfiles=10,redir=""))
 
     # pre-defined tests
-    mytests.append(Test("Summer20UL16","TTToHadronic_TuneCP5_13TeV-powheg-pythia8",numevents,command,inputFilesConfig="Summer20UL16.TTToHadronic_TuneCP5_13TeV-powheg-pythia8",nstart=0,nfiles=1))
-    mytests.append(Test("Summer20UL16APV","TTToHadronic_TuneCP5_13TeV-powheg-pythia8",numevents,command,inputFilesConfig="Summer20UL16APV.TTToHadronic_TuneCP5_13TeV-powheg-pythia8",nstart=0,nfiles=1))
-    mytests.append(Test("Summer20UL17","TTToHadronic_TuneCP5_13TeV-powheg-pythia8",numevents,command,inputFilesConfig="Summer20UL17.TTToHadronic_TuneCP5_13TeV-powheg-pythia8",nstart=0,nfiles=1))
-    mytests.append(Test("Summer20UL18","TTToHadronic_TuneCP5_13TeV-powheg-pythia8",numevents,command,inputFilesConfig="Summer20UL18.TTToHadronic_TuneCP5_13TeV-powheg-pythia8",nstart=0,nfiles=1))
+    mytests.append(Test("Summer20UL16","TTToSemiLeptonic_TuneCP5_13TeV-powheg-pythia8",numevents,command,inputFilesConfig="Summer20UL16.TTToSemiLeptonic_TuneCP5_13TeV-powheg-pythia8",nstart=0,nfiles=1))
```
Comment (kpedro88) [inline Production/test/unitTest.py]: This will conflict with #592

File: Production/test/unitTest.py [hunk 3]
```diff
@@ -93,12 +93,22 @@ def defineTests(mytests, scenario, name, numevents, command, dataset, inputFiles
...
+    mytests.append(Test("Summer20UL16APV","TTToSemiLeptonic_TuneCP5_13TeV-powheg-pythia8",numevents,command,inputFilesConfig="Summer20UL16APV.TTToSemiLeptonic_TuneCP5_13TeV-powheg-pythia8",nstart=0,nfiles=1))
+    mytests.append(Test("Summer20UL17","TTToSemiLeptonic_TuneCP5_13TeV-powheg-pythia8",numevents,command,inputFilesConfig="Summer20UL17.TTToSemiLeptonic_TuneCP5_13TeV-powheg-pythia8",nstart=0,nfiles=1))
+    mytests.append(Test("Summer20UL18","TTToSemiLeptonic_TuneCP5_13TeV-powheg-pythia8",numevents,command,inputFilesConfig="Summer20UL18.TTToSemiLeptonic_TuneCP5_13TeV-powheg-pythia8",nstart=0,nfiles=1))
     mytests.append(Test("Summer20UL16","GJets_HT-100To200_TuneCP5_13TeV-madgraphMLM-pythia8",numevents,command,inputFilesConfig="Summer20UL16.GJets_HT-100To200_TuneCP5_13TeV-madgraphMLM-pythia8",nstart=0,nfiles=1))
     mytests.append(Test("Summer20UL16APV","GJets_HT-100To200_TuneCP5_13TeV-madgraphMLM-pythia8",numevents,command,inputFilesConfig="Summer20UL16APV.GJets_HT-100To200_TuneCP5_13TeV-madgraphMLM-pythia8",nstart=0,nfiles=1))
+    mytests.append(Test("Summer20UL16_DATA",name,numevents,command,inputFilesConfig="Run2016G-UL2016-v2.JetHT",nstart=0,nfiles=1))
```
Comment (kpedro88) [inline Production/test/unitTest.py]: `name` should be replaced with the correct output filename for each test

File: Production/test/unitTest.py [hunk 4]
```diff
@@ -93,12 +93,22 @@ def defineTests(mytests, scenario, name, numevents, command, dataset, inputFiles
...
+    mytests.append(Test("Summer20UL16_DATA",name,numevents,command,inputFilesConfig="Run2016H-UL2016-v2.JetHT",nstart=0,nfiles=1))
+    mytests.append(Test("Summer20UL16APV_DATA",name,numevents,command,inputFilesConfig="Run2016D-UL2016_HIPM-v2.SingleElectron",nstart=0,nfiles=1))
+    mytests.append(Test("Summer20UL17_DATA",name,numevents,command,inputFilesConfig="Run2017E-UL2017-v1.SingleMuon",nstart=0,nfiles=1))
+    mytests.append(Test("Summer20UL18_DATA",name,numevents,command,inputFilesConfig="Run2018A-UL2018-v2.MET",nstart=0,nfiles=1))
+    mytests.append(Test("Summer20UL18_DATA",name,numevents,command,inputFilesConfig="Run2018C-UL2018-v1.EGamma",nstart=0,nfiles=1))
+    mytests.append(Test("Summer20UL16","GJets_HT-100To200_TuneCP5_13TeV-madgraphMLM-pythia8",numevents,command,inputFilesConfig="Summer20UL16.GJets_HT-100To200_TuneCP5_13TeV-madgraphMLM-pythia8",nstart=0,nfiles=1))
```
Comment (kpedro88) [inline Production/test/unitTest.py]: redundant w/ line 100 above?

File: Production/test/unitTest.py [hunk 5]
```diff
@@ -93,12 +93,22 @@ def defineTests(mytests, scenario, name, numevents, command, dataset, inputFiles
...
+    mytests.append(Test("Summer20UL16APV","GJets_HT-100To200_TuneCP5_13TeV-madgraphMLM-pythia8",numevents,command,inputFilesConfig="Summer20UL16APV.GJets_HT-100To200_TuneCP5_13TeV-madgraphMLM-pythia8",nstart=0,nfiles=1))
+    mytests.append(Test("Summer20UL17","GJets_DR-0p4_HT-100To200_TuneCP5_13TeV-madgraphMLM-pythia8",numevents,command,inputFilesConfig="Summer20UL17.GJets_DR-0p4_HT-100To200_TuneCP5_13TeV-madgraphMLM-pythia8",nstart=0,nfiles=1))
```
Comment (kpedro88) [inline Production/test/unitTest.py]: in general, please follow the naming convention where the year/era is included in the filename (see #592)

File: Production/test/unitTest.py (hunk 2 above)
Comment (JHiltbrand) [reply Production/test/unitTest.py]: Ah, I see that now. Thanks for pointing that out.
